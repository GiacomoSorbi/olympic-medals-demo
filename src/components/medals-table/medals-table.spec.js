import * as React from 'react'
import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'

import MedalsTable from './'

describe('MedalsTable should', () => {
  it('exist', () => {
    expect(MedalsTable).not.toBeUndefined()
  })

  it('have a consistent structure', () => {
    const wrapper = mount(
      <MedalsTable
        sortCriteria='bronze-desc'
        data={[]}
        changeSorting={jest.fn()}
        onEdit={jest.fn()}
      />,
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
