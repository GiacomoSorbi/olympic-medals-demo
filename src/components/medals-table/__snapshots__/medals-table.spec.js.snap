// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MedalsTable should have a consistent structure 1`] = `
<MedalsTable
  changeSorting={[MockFunction]}
  data={Array []}
  onEdit={[MockFunction]}
  sortCriteria="bronze-desc"
>
  <div
    className="table-container mx-auto my-5"
  >
    <img
      alt="logo"
      className="olympic-logo"
      src="olympic-logo.svg"
    />
    <div
      className="table-responsive border border-secondary"
    >
      <Table
        responsiveTag="div"
        tag="table"
      >
        <table
          className="table"
        >
          <thead>
            <tr
              className="light"
            >
              <th
                className="align-middle border"
              >
                Country
              </th>
              <th
                className="align-middle border"
                key="gold"
              >
                <div
                  className="d-flex align-items-center justify-content-between"
                >
                  <Medal
                    color="gold"
                  >
                    <svg
                      className="medal-icon"
                      id="Capa_1"
                      style={
                        Object {
                          "enableBackground": "new 0 0 297.334 297.334",
                        }
                      }
                      version="1.1"
                      viewBox="0 0 297.334 297.334"
                      x="0px"
                      xmlns="http://www.w3.org/2000/svg"
                      y="0px"
                    >
                      <g>
                        <polygon
                          points="172.667,64 206.687,0 90.623,0 124.649,64  "
                        />
                        <path
                          d="M227.904,8.293c-2.553-4.837-6.726-4.834-9.272,0.006l-55.839,106.095c20.009,2.892,38.11,11.76,52.427,24.735   l36.208-68.682c2.551-4.838,2.549-12.754-0.004-17.591L227.904,8.293z"
                        />
                        <path
                          d="M134.518,114.35L78.7,8.299c-2.547-4.84-6.72-4.843-9.272-0.006L45.909,52.856c-2.552,4.837-2.554,12.753-0.004,17.591   l36.102,68.48C96.362,125.996,114.491,117.181,134.518,114.35z"
                        />
                        <path
                          d="M148.502,128.279c-46.685,0-84.529,37.844-84.529,84.528c0,46.682,37.845,84.526,84.529,84.526   c46.682,0,84.526-37.845,84.526-84.526C233.028,166.123,195.184,128.279,148.502,128.279z M148.502,273.125   c-33.262,0-60.32-27.059-60.32-60.317c0-33.261,27.059-60.319,60.32-60.319c33.259,0,60.316,27.059,60.316,60.319   C208.819,246.066,181.761,273.125,148.502,273.125z"
                          fill="#ffd700"
                        />
                        <path
                          d="M148.502,167.404c-25.038,0-45.404,20.367-45.404,45.403c0,25.033,20.366,45.4,45.404,45.4   c25.033,0,45.399-20.367,45.399-45.4C193.902,187.771,173.535,167.404,148.502,167.404z"
                          fill="#ffd700"
                        />
                      </g>
                    </svg>
                  </Medal>
                  <div
                    className="no-mobile"
                  >
                    Gold
                     Medals
                  </div>
                  <SortButtons
                    color="gold"
                    onClick={[MockFunction]}
                    sortCriteria="bronze-desc"
                  >
                    <div
                      className="sort-buttons d-flex flex-column"
                    >
                      <Button
                        color="primary"
                        data-sort-direction="gold-desc"
                        onClick={[MockFunction]}
                        tag="button"
                      >
                        <button
                          aria-label={null}
                          className="btn btn-primary"
                          data-sort-direction="gold-desc"
                          onClick={[Function]}
                          type="button"
                        >
                          <img
                            alt="edit row"
                            src="arrow.svg"
                          />
                        </button>
                      </Button>
                      <Button
                        className="reversed"
                        color="primary"
                        data-sort-direction="gold-asc"
                        onClick={[MockFunction]}
                        tag="button"
                      >
                        <button
                          aria-label={null}
                          className="reversed btn btn-primary"
                          data-sort-direction="gold-asc"
                          onClick={[Function]}
                          type="button"
                        >
                          <img
                            alt="edit row"
                            src="arrow.svg"
                          />
                        </button>
                      </Button>
                    </div>
                  </SortButtons>
                </div>
              </th>
              <th
                className="align-middle border"
                key="silver"
              >
                <div
                  className="d-flex align-items-center justify-content-between"
                >
                  <Medal
                    color="silver"
                  >
                    <svg
                      className="medal-icon"
                      id="Capa_1"
                      style={
                        Object {
                          "enableBackground": "new 0 0 297.334 297.334",
                        }
                      }
                      version="1.1"
                      viewBox="0 0 297.334 297.334"
                      x="0px"
                      xmlns="http://www.w3.org/2000/svg"
                      y="0px"
                    >
                      <g>
                        <polygon
                          points="172.667,64 206.687,0 90.623,0 124.649,64  "
                        />
                        <path
                          d="M227.904,8.293c-2.553-4.837-6.726-4.834-9.272,0.006l-55.839,106.095c20.009,2.892,38.11,11.76,52.427,24.735   l36.208-68.682c2.551-4.838,2.549-12.754-0.004-17.591L227.904,8.293z"
                        />
                        <path
                          d="M134.518,114.35L78.7,8.299c-2.547-4.84-6.72-4.843-9.272-0.006L45.909,52.856c-2.552,4.837-2.554,12.753-0.004,17.591   l36.102,68.48C96.362,125.996,114.491,117.181,134.518,114.35z"
                        />
                        <path
                          d="M148.502,128.279c-46.685,0-84.529,37.844-84.529,84.528c0,46.682,37.845,84.526,84.529,84.526   c46.682,0,84.526-37.845,84.526-84.526C233.028,166.123,195.184,128.279,148.502,128.279z M148.502,273.125   c-33.262,0-60.32-27.059-60.32-60.317c0-33.261,27.059-60.319,60.32-60.319c33.259,0,60.316,27.059,60.316,60.319   C208.819,246.066,181.761,273.125,148.502,273.125z"
                          fill="#c0c0c0"
                        />
                        <path
                          d="M148.502,167.404c-25.038,0-45.404,20.367-45.404,45.403c0,25.033,20.366,45.4,45.404,45.4   c25.033,0,45.399-20.367,45.399-45.4C193.902,187.771,173.535,167.404,148.502,167.404z"
                          fill="#c0c0c0"
                        />
                      </g>
                    </svg>
                  </Medal>
                  <div
                    className="no-mobile"
                  >
                    Silver
                     Medals
                  </div>
                  <SortButtons
                    color="silver"
                    onClick={[MockFunction]}
                    sortCriteria="bronze-desc"
                  >
                    <div
                      className="sort-buttons d-flex flex-column"
                    >
                      <Button
                        color="primary"
                        data-sort-direction="silver-desc"
                        onClick={[MockFunction]}
                        tag="button"
                      >
                        <button
                          aria-label={null}
                          className="btn btn-primary"
                          data-sort-direction="silver-desc"
                          onClick={[Function]}
                          type="button"
                        >
                          <img
                            alt="edit row"
                            src="arrow.svg"
                          />
                        </button>
                      </Button>
                      <Button
                        className="reversed"
                        color="primary"
                        data-sort-direction="silver-asc"
                        onClick={[MockFunction]}
                        tag="button"
                      >
                        <button
                          aria-label={null}
                          className="reversed btn btn-primary"
                          data-sort-direction="silver-asc"
                          onClick={[Function]}
                          type="button"
                        >
                          <img
                            alt="edit row"
                            src="arrow.svg"
                          />
                        </button>
                      </Button>
                    </div>
                  </SortButtons>
                </div>
              </th>
              <th
                className="align-middle border"
                key="bronze"
              >
                <div
                  className="d-flex align-items-center justify-content-between"
                >
                  <Medal
                    color="bronze"
                  >
                    <svg
                      className="medal-icon"
                      id="Capa_1"
                      style={
                        Object {
                          "enableBackground": "new 0 0 297.334 297.334",
                        }
                      }
                      version="1.1"
                      viewBox="0 0 297.334 297.334"
                      x="0px"
                      xmlns="http://www.w3.org/2000/svg"
                      y="0px"
                    >
                      <g>
                        <polygon
                          points="172.667,64 206.687,0 90.623,0 124.649,64  "
                        />
                        <path
                          d="M227.904,8.293c-2.553-4.837-6.726-4.834-9.272,0.006l-55.839,106.095c20.009,2.892,38.11,11.76,52.427,24.735   l36.208-68.682c2.551-4.838,2.549-12.754-0.004-17.591L227.904,8.293z"
                        />
                        <path
                          d="M134.518,114.35L78.7,8.299c-2.547-4.84-6.72-4.843-9.272-0.006L45.909,52.856c-2.552,4.837-2.554,12.753-0.004,17.591   l36.102,68.48C96.362,125.996,114.491,117.181,134.518,114.35z"
                        />
                        <path
                          d="M148.502,128.279c-46.685,0-84.529,37.844-84.529,84.528c0,46.682,37.845,84.526,84.529,84.526   c46.682,0,84.526-37.845,84.526-84.526C233.028,166.123,195.184,128.279,148.502,128.279z M148.502,273.125   c-33.262,0-60.32-27.059-60.32-60.317c0-33.261,27.059-60.319,60.32-60.319c33.259,0,60.316,27.059,60.316,60.319   C208.819,246.066,181.761,273.125,148.502,273.125z"
                          fill="#cd7f32"
                        />
                        <path
                          d="M148.502,167.404c-25.038,0-45.404,20.367-45.404,45.403c0,25.033,20.366,45.4,45.404,45.4   c25.033,0,45.399-20.367,45.399-45.4C193.902,187.771,173.535,167.404,148.502,167.404z"
                          fill="#cd7f32"
                        />
                      </g>
                    </svg>
                  </Medal>
                  <div
                    className="no-mobile"
                  >
                    Bronze
                     Medals
                  </div>
                  <SortButtons
                    color="bronze"
                    onClick={[MockFunction]}
                    sortCriteria="bronze-desc"
                  >
                    <div
                      className="sort-buttons d-flex flex-column"
                    >
                      <Button
                        color="success"
                        data-sort-direction="bronze-desc"
                        onClick={[MockFunction]}
                        tag="button"
                      >
                        <button
                          aria-label={null}
                          className="btn btn-success"
                          data-sort-direction="bronze-desc"
                          onClick={[Function]}
                          type="button"
                        >
                          <img
                            alt="edit row"
                            src="arrow.svg"
                          />
                        </button>
                      </Button>
                      <Button
                        className="reversed"
                        color="primary"
                        data-sort-direction="bronze-asc"
                        onClick={[MockFunction]}
                        tag="button"
                      >
                        <button
                          aria-label={null}
                          className="reversed btn btn-primary"
                          data-sort-direction="bronze-asc"
                          onClick={[Function]}
                          type="button"
                        >
                          <img
                            alt="edit row"
                            src="arrow.svg"
                          />
                        </button>
                      </Button>
                    </div>
                  </SortButtons>
                </div>
              </th>
              <th
                className="align-middle border"
                key="total"
              >
                <div
                  className="d-flex align-items-center justify-content-between"
                >
                  <Medal
                    color="total"
                  >
                    <svg
                      className="medal-icon"
                      id="Capa_1"
                      style={
                        Object {
                          "enableBackground": "new 0 0 297.334 297.334",
                        }
                      }
                      version="1.1"
                      viewBox="0 0 297.334 297.334"
                      x="0px"
                      xmlns="http://www.w3.org/2000/svg"
                      y="0px"
                    >
                      <g>
                        <polygon
                          points="172.667,64 206.687,0 90.623,0 124.649,64  "
                        />
                        <path
                          d="M227.904,8.293c-2.553-4.837-6.726-4.834-9.272,0.006l-55.839,106.095c20.009,2.892,38.11,11.76,52.427,24.735   l36.208-68.682c2.551-4.838,2.549-12.754-0.004-17.591L227.904,8.293z"
                        />
                        <path
                          d="M134.518,114.35L78.7,8.299c-2.547-4.84-6.72-4.843-9.272-0.006L45.909,52.856c-2.552,4.837-2.554,12.753-0.004,17.591   l36.102,68.48C96.362,125.996,114.491,117.181,134.518,114.35z"
                        />
                        <path
                          d="M148.502,128.279c-46.685,0-84.529,37.844-84.529,84.528c0,46.682,37.845,84.526,84.529,84.526   c46.682,0,84.526-37.845,84.526-84.526C233.028,166.123,195.184,128.279,148.502,128.279z M148.502,273.125   c-33.262,0-60.32-27.059-60.32-60.317c0-33.261,27.059-60.319,60.32-60.319c33.259,0,60.316,27.059,60.316,60.319   C208.819,246.066,181.761,273.125,148.502,273.125z"
                          fill="#8F8"
                        />
                        <path
                          d="M148.502,167.404c-25.038,0-45.404,20.367-45.404,45.403c0,25.033,20.366,45.4,45.404,45.4   c25.033,0,45.399-20.367,45.399-45.4C193.902,187.771,173.535,167.404,148.502,167.404z"
                          fill="#8F8"
                        />
                      </g>
                    </svg>
                  </Medal>
                  <div
                    className="no-mobile"
                  >
                    Total
                     Medals
                  </div>
                  <SortButtons
                    color="total"
                    onClick={[MockFunction]}
                    sortCriteria="bronze-desc"
                  >
                    <div
                      className="sort-buttons d-flex flex-column"
                    >
                      <Button
                        color="primary"
                        data-sort-direction="total-desc"
                        onClick={[MockFunction]}
                        tag="button"
                      >
                        <button
                          aria-label={null}
                          className="btn btn-primary"
                          data-sort-direction="total-desc"
                          onClick={[Function]}
                          type="button"
                        >
                          <img
                            alt="edit row"
                            src="arrow.svg"
                          />
                        </button>
                      </Button>
                      <Button
                        className="reversed"
                        color="primary"
                        data-sort-direction="total-asc"
                        onClick={[MockFunction]}
                        tag="button"
                      >
                        <button
                          aria-label={null}
                          className="reversed btn btn-primary"
                          data-sort-direction="total-asc"
                          onClick={[Function]}
                          type="button"
                        >
                          <img
                            alt="edit row"
                            src="arrow.svg"
                          />
                        </button>
                      </Button>
                    </div>
                  </SortButtons>
                </div>
              </th>
              <th
                className="align-middle border"
              >
                Edit
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
      </Table>
    </div>
  </div>
</MedalsTable>
`;
